src/main/java/com/connexta/store/poller/TransformStatusPoller.java

```diff
@@ -0,0 +1,281 @@
+
+  private final WebClient transformWebClient;
+
+  private final BlockingQueue<TransformStatusTask> transformStatusQueue;
+
+  /** Creates a new TransformStatusPoller. */
+  public TransformStatusPoller(
+      BlockingQueue<TransformStatusTask> transformStatusQueue,
+      WebClient transformWebClient,
+      WebClient storeWebClient) {
+    this.transformStatusQueue = transformStatusQueue;
+    this.transformWebClient = transformWebClient;
+    this.storeWebClient = storeWebClient;
+  }
+
+  // TODO 11/20/2019 PeterHuffer: This class should be managed and be able to be properly destroyed
+  @SuppressWarnings("squid:S2189" /* Run forever by design */)
+  @Override
+  public void run() {
+    while (true) {
+      try {
+        doRun();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        return;
+      }
+    }
+  }
+
+  @VisibleForTesting
+  void doRun() throws InterruptedException {
+    TransformStatusTask transformStatusTask = transformStatusQueue.take();
+    boolean shouldRetry = handleTask(transformStatusTask);
+    if (shouldRetry) {
+      boolean resubmitted = transformStatusQueue.offer(transformStatusTask, 10, TimeUnit.SECONDS);
+      if (resubmitted) {
+        log.debug(
+            "Transform status {} will be tried again for dataset {}.",
+            transformStatusTask.getTransformStatusUrl(),
+            transformStatusTask.getDatasetId());
+      } else {
+        log.debug(
+            "Timed out trying to resubmit transform status {} for dataset {}. The file corresponding to this dataset should be reingsted.",
+            transformStatusTask.getTransformStatusUrl(),
+            transformStatusTask.getDatasetId());
+      }
+    }
+  }
+
+  /** @return {@code true} if the task should be retried, otherwise {@code false} */
+  private boolean handleTask(TransformStatusTask transformStatusTask) {
+    final String transformPollUrl = transformStatusTask.getTransformStatusUrl().toString();
+    final String datasetId = transformStatusTask.getDatasetId();
+
+    // this could be an ErrorMessage too, need to account for that
+    ClientResponse response = transformWebClient.get().uri(transformPollUrl).exchange().block();
+
+    if (response != null) {
+      final HttpStatus httpStatus = response.statusCode();
+
+      if (HttpStatus.OK.equals(httpStatus)) {
+        ResponseEntity<TransformationPollResponse> transformationPollResponse =
+            response.toEntity(TransformationPollResponse.class).block();
+        if (transformationPollResponse != null) {
+          if (transformationPollResponse.getBody() == null) {
+            log.debug(
+                "No body available in the successful transform status response for status {}. This URL will be retried.",
+                transformPollUrl);
+            return true;
+          }
+
+          return handlePollResponse(datasetId, transformationPollResponse.getBody());
+        } else {
+          log.debug(
```

Comment:
✏️ It looks like this log could be combined with the one above
```        
        if (transformationPollResponse != null && transformationPollResponse.getBody() != null) {
            return handlePollResponse(datasetId, transformationPollResponse.getBody());
          }
        log.debug(
                "No body available in the successful transform status response for status {}. This URL will be retried.", 
                transformPollUrl);
        return true;

Reply:
Whoops, that second log message should be different. Bad copy paste.